src/command_modules/azure-cli-hdinsight/azure/cli/command_modules/hdinsight/custom.py

```diff
@@ -0,0 +1,238 @@
+                            target_instance_count=3,
+                            hardware_profile=HardwareProfile(vm_size=zookeepernode_size),
+                            os_profile=os_profile,
+                            virtual_network_profile=virtual_network_profile
+                        )
+                    ] if zookeepernode_size else []
+                ) +
+                (
+                    [
+                        # Specify an edgenode role only if the edgenode size is specified
+                        Role(
+                            name="edgenode",
+                            target_instance_count=1,
+                            hardware_profile=HardwareProfile(vm_size=edgenode_size),
+                            os_profile=os_profile,
+                            virtual_network_profile=virtual_network_profile
+                        )
+                    ] if edgenode_size else []
+                )
+            ),
+            storage_profile=StorageProfile(
+                storageaccounts=(
+                    [
+                        # Specify storage account details only when storage arguments are provided
+                        StorageAccount(
+                            name=storage_account,
+                            key=storage_account_key,
+                            container=storage_default_container,
+                            file_system=storage_default_filesystem,
+                            is_default=True
+                        )
+                    ] if storage_account else []
+                ) +
+                [
+                    StorageAccount(
+                        name=s.storage_account,
+                        key=s.storage_account_key,
+                        container=s.container,
+                        is_default=False
+                    )
+                    for s in additional_storage_accounts
+                ]
+            )
+        )
+    )
+
+    if no_wait:
+        return sdk_no_wait(no_wait, client.create,
+                           resource_group_name, cluster_name, create_params)
+
+    return client.create(resource_group_name, cluster_name, create_params)
+
+
+def list_clusters(cmd, client, resource_group_name=None):  # pylint: disable=unused-argument
+    clusters_list = client.list_by_resource_group(resource_group_name=resource_group_name) \
+        if resource_group_name else client.list()
+
+    return list(clusters_list)
+
+
+def _all_or_none(*params):
+    return all(params) or not any(params)
+
+
+def _get_rg_location(ctx, resource_group_name, subscription_id=None):
+    from ._client_factory import cf_resource_groups
+    groups = cf_resource_groups(ctx, subscription_id=subscription_id)
+    # Just do the get, we don't need the result, it will error out if the group doesn't exist.
+    rg = groups.get(resource_group_name)
+    return rg.location
+
+
+def _is_wasb_endpoint(storage_endpoint):
+    return '.blob.' in storage_endpoint
```

Comment:
This may not be stable @williexu

Reply:
I'm open to suggestions. One thing that will make this difficult is that the user might not have resource access to this storage account, and it could also be in a different subscription than the resource being created.